test/backend/QueryScheduler.spec.js

```diff
@@ -114,11 +114,161 @@ describe('QueryScheduler', function() {
+        const clock = sinon.useFakeTimers();
+        const spy = sinon.spy();
+        const ONE_WEEK = 604800;
+        const cronInterval = '26 43 14 * * 5'; // arbitrary day and time of week
+
+        queryScheduler.job = spy;
+        queryScheduler.scheduleQuery({
+            cronInterval,
+            fid: '...',
+            uids: '...',
+            query: '...',
+            connectionId: '...'
+        });
+
+        assert(spy.notCalled, 'job should not have been called yet');
+
+        clock.tick(ONE_WEEK * 1000);
+        assert(spy.called, 'job should have been called');
+        clock.tick(ONE_WEEK * 1000);
+        assert(spy.calledTwice, 'job should have been called twice');
+
+        clock.restore();
+    });
+
+    it('cronInteval overrules refreshInterval', function () {
+        const clock = sinon.useFakeTimers();
+        const spy = sinon.spy();
+
+        const refreshInterval = 60; // one minute
+
+        const date = new Date();
+        date.setMinutes(date.getMinutes() + 5);
+        const fiveMinutesFromNow = date.getMinutes();
+        const cronInterval = `${fiveMinutesFromNow} * * * *`;
+
+        queryScheduler.job = spy;
+        queryScheduler.scheduleQuery({
+            cronInterval,
+            refreshInterval,
+            fid: '...',
+            uids: '...',
+            query: '...',
+            connectionId: '...'
+        });
+
+        clock.tick(1.5 * refreshInterval * 1000);
+        assert(spy.notCalled, 'job should not have been called yet');
+        clock.tick(4 * refreshInterval * 1000);
+        assert(spy.calledOnce, 'job should have been called once');
+
+        clock.restore();
+    });
+
+    it('correctly maps refreshIntervals to cronIntervals', function () {
+        const clock = sinon.useFakeTimers();
+        let testCount = 0;
+
+        function checkCase (refreshInterval, message) {
+            const date = new Date();
+            date.setSeconds(date.getSeconds() + refreshInterval);
+            const futureTimeTarget = date.toISOString();
+
+            queryScheduler.scheduleQuery({
+                refreshInterval: refreshInterval,
+                fid: `test:${++testCount}`,
+                uids: '...',
+                query: '...',
+                connectionId: '...'
+            });
+            const scheduledRuntime = queryScheduler.queryJobs[`test:${testCount}`].nextInvocation().toISOString();
+            assert.equal(scheduledRuntime, futureTimeTarget, message);
+        }
+
+        // test against each UI possibilty
+        checkCase(60, 'refreshInterval of 60 should run one minute later');
+        checkCase(300, 'refreshInterval of 300 should run five minutes later');
+        checkCase(3600, 'refreshInterval of 3600 should run one hour later');
+        checkCase(86400, 'refreshInterval of 86400 should run one day later');
+        checkCase(604800, 'refreshInterval of 604800 should run one week later');
+
+        clock.reset();
```

Comment:
This is the line that's causing the test failure in CircleCI. Replace with `clock.restore()`.